services/app-web/src/hooks/useContractForm.ts

```diff
@@ -0,0 +1,260 @@
+import { useState, useEffect} from 'react'
+import { usePage } from "../contexts/PageContext"
+import {
+    CreateContractInput,
+    useFetchContractQuery,
+    useCreateContractMutation,
+    useUpdateContractDraftRevisionMutation,
+    Contract,
+    Rate,
+    GenericDocument,
+    GenericDocumentInput,
+    StateContactInput,
+    StateContact,
+    UnlockedContract,
+    UpdateContractDraftRevisionInput,
+    ContractPackageSubmission
+} from '../gen/gqlClient'
+import { recordJSException } from '../otelHelpers'
+import { handleApolloError } from '../gqlHelpers/apolloErrors'
+import { ApolloError } from '@apollo/client'
+import type { InterimState } from '../pages/StateSubmission/ErrorOrLoadingPage'
+
+
+type  UseContractForm = {
+    draftSubmission?: UnlockedContract
+    showPageErrorMessage: string | boolean
+    previousDocuments?: string[]
+    updateDraft: (
+        input: UpdateContractDraftRevisionInput
+    ) => Promise<Contract | Error>
+    createDraft: (input: CreateContractInput) => Promise<Contract | Error>
+    interimState?:  InterimState
+}
+
+const documentsInput = (documents: GenericDocument[]): GenericDocumentInput[] => {
+    return documents.map((doc) => {
+        return {
+            downloadURL: doc.downloadURL,
+            name: doc.name,
+            s3URL: doc.s3URL,
+            sha256: doc.sha256
+        }
+    })
+}
+
+const stateContactsInput = (contacts: StateContact[]): StateContactInput[] => {
+    return contacts.map((contact) => {
```

Comment:
Is the point of these input functions mainly to strip out `__typename `? 

For cases where that is the only difference -  could something like [this ](l https://www.apollographql.com/docs/react/api/link/apollo-link-remove-typename/#remove-__typename-from-all-variables ) be useful. Starting to wonder with moving to graphql types everywhere might it be better to do graphql level config change?

This is not something we have to implement now don't want to derail your work,  raising it more for discussion.

Reply:
Thanks! I meant to call this out as well because I felt like there had to be another way